src/fs/fd.rs

```diff
@@ -161,6 +162,7 @@ pub fn fchown<Fd: AsFd>(fd: Fd, owner: Option<Uid>, group: Option<Gid>) -> io::R
 /// [Linux]: https://man7.org/linux/man-pages/man2/fstat.2.html
 /// [`Mode::from_raw_mode`]: Mode::from_raw_mode
 /// [`FileType::from_raw_mode`]: crate::fs::FileType::from_raw_mode
+#[cfg(not(target_arch = "loongarch64"))]
 #[inline]
 pub fn fstat<Fd: AsFd>(fd: Fd) -> io::Result<Stat> {
```

Comment:
I assume loongarch64 should support the public `fstat` API. There may be no `__NR_fstat` but if so then src/backend/linux_raw/fs/syscalls.rs' `fstat` can probably use `statx` instead.